fhir-persistence-schema/docs/DB2MultiTenancy.md

```diff
@@ -0,0 +1,140 @@
+
+Each tenant's data is able to be moved, backedup, restored or exported a tenant-at-a-time.  This approach allows tenant independent restore points.
+
+The isolated data supports tenant independent query plan caches, scaling and disambiguating tenant specific disk and space usage. The performance metrics are tenant specific, and diagnostics are tenant specific. 
+
+**Table: Resources**
+
+Each resource is contains a root RESOURCE and LOGICAL RESOURCE, such as VISIONPRESCRIPTION:
+
+```
+VISIONPRESCRIPTION_RESOURCES
+VISIONPRESCRIPTION_LOGICAL_RESOURCES
+```
+
+These values are mapped to specific FHIR Search value types, supporting parameter searches: 
+
+```
+VISIONPRESCRIPTION_COMPOSITES
+VISIONPRESCRIPTION_DATE_VALUES
+VISIONPRESCRIPTION_LATLNG_VALUES
+VISIONPRESCRIPTION_NUMBER_VALUES
+VISIONPRESCRIPTION_QUANTITY_VALUES
+VISIONPRESCRIPTION_STR_VALUES
+VISIONPRESCRIPTION_TOKEN_VALUES
+```
+
+There are also global tables which enable a global search using the logical_resource and resource.  
+
+Each row is secured using DB2 Row Permission. The Stored Procedure is the ONLY way for a user to set the variable, which is used in the DB2 Row Permission. 
+
+Each tenant's data is encrypted. 
+
+Offboarding a tenant's data is accomplished using dettach/drop partition, and subsequently dropping the tablespace. 
+
+**Stored Procedure: FHIR_ADMIN.SET_TENANT**
+
+The FHIR_ADMIN.SET_TENANT is used at the beginning of a connection to the IBM FHIR Server's multi-tenant schema. The SET_TENANT takes as input: tenant_name VARCHAR(36), tenant_key  VARCHAR(44).  The content outputs a connection specific variable `fhir_admin.sv_tenant_id` with the `tenant.mt_id`, which is automatically applied to the INSERT,SELECT,UPDATE,DELETE.  Only tenant's with status ALLOCATED are allowed to execute the actions.
+
+To use the stored procedure from the db2 administrative console. 
+
+* Map Tenant and Key to Tenant ID (Stored Procedure) - `CALL FHIR_ADMIN.set_tenant('a-tenant-name', 'BLAH_BLAH');`
+* Show the value set for the session - `VALUES fhir_admin.sv_tenant_id;`
+* Execute the query, such as `SELECT * FROM FHIRDATA.MEDICATION_RESOURCES`
+
+**Schema: Permissions**
+
+The Multi-tenant schema sets the following permissions: 
+- All objects in schema FHIRADMIN are owned by user FHIRADMIN
+- All objects in schema FHIRDATA are owned by user FHIRADMIN
+- GRANT EXECUTE ON FHIRADMIN.SET_TENANT TO FHIRUSER;
+- GRANT READ ON FHIRADMIN.SV_TENANT_ID TO FHIRUSER;
+- GRANT EXECUTE ON FHIRDATA.ADD_RESOURCE_TYPE TO FHIRUSER;
+- GRANT EXECUTE ON FHIRDATA.ADD_RESOURCE TO FHIRUSER;
+- GRANT SELECT,INSERT,UPDATE,DELETE ON FHIRDATA.PATIENT_RESOURCES TO FHIRUSER;
+
+FHIRADMIN owns all schema objects. 
+FHIR User is granted a minimum set of priveleges. 
+SET_TENANT is the only way to write the SV_TENANT_ID. 
+
+The diagram outlines the relationships between the administrative and tenant schema. 
+![permissions-overview.png](permissions-overview.png)
```

Comment:
please update diagram to use `FHIRSERVER` instead of `FHIRUSER` as the db user

Reply:
done